service/migration_manager.cc

```diff
@@ -154,11 +154,12 @@ void migration_manager::init_messaging_service()
         return make_ready_future<utils::UUID>(service::get_local_storage_proxy().get_db().local().get_version());
     });
     _messaging.register_get_schema_version([this] (unsigned shard, table_schema_version v) {
-        get_local_storage_proxy().get_stats().replica_cross_shard_ops += shard != this_shard_id();
+        auto& proxy = get_local_storage_proxy();
+        proxy.get_stats().replica_cross_shard_ops += shard != this_shard_id();
         // FIXME: should this get an smp_service_group? Probably one separate from reads and writes.
-        return container().invoke_on(shard, [v] (auto&& sp) {
+        return container().invoke_on(shard, [v, &proxy] (auto&& sp) {
             mlogger.debug("Schema version request for {}", v);
-            return local_schema_registry().get_frozen(v);
+            return proxy.local_db().get_schema_registry().get_frozen(v);
```

Comment:
This is dangerous. The proxy reference is carried over to an alien shard, then used there. In safe manner but still. Better capture the `&db = proxy.local_db()`.

Reply:
Just curious - is capturing  an alien database reference for some reason better than capturing storage proxy? Perhaps because the database has better liveness guarantees?

Reply:
Ah ok, we explicitly **want** to use a foreign database here anyway. I guess I understand the comment now